File: Sample.kt
import java.util.Scanner
class MatrixSum(private val r: Int, private val c: Int) {
    private val x = Array(r) { IntArray(c) }
    private val y = Array(r) { IntArray(c) }
    private val z = Array(r) { IntArray(c) }

    private val sc = Scanner(System.in)
    fun input() {
        println("Enter elements of the 1st Matrix:")
        for (i in 0 until r) {
            for (j in 0 until c) {
                x[i][j] = sc.nextInt()
            }
        }
        println("Enter elements of the 2nd Matrix:")
        for (i in 0 until r) {
            for (j in 0 until c) {
                y[i][j] = sc.nextInt()
            }
        }
    }
    fun calculate() {
        for (i in 0 until r) {
            for (j in 0 until c) {
                z[i][j] = x[i][j] + y[i][j]
            }
        }
    }
    fun display() {
        println("Output Matrix:")
        for (i in 0 until r) {
            for (j in 0 until c) {
                print("${z[i][j]}\t")
            }
            println()
        }
    }
}
fun main() {
    val sc = Scanner(System.in)

    println("Enter the number of rows and columns of the matrices:")
    val r = sc.nextInt()
    val c = sc.nextInt()
    if (r > 0 && c > 0) {
        val obj = MatrixSum(r, c)
        obj.input()
        obj.calculate()
        obj.display()
    } else {
        println("Error: Rows and columns should be greater than 0")
    }
}

File: Sample.c
#include <stdio.h>

void input(int r, int c, int matrix[r][c], const char* name) {
    printf("Enter elements of the %s Matrix:\n", name);
    for (int i = 0; i < r; i++) {
        for (int j = 0; j < c; j++) {
            scanf("%d", &matrix[i][j]);
        }
    }
}

void calculate(int r, int c, int matrix1[r][c], int matrix2[r][c], int result[r][c]) {
    for (int i = 0; i < r; i++) {
        for (int j = 0; j < c; j++) {
            result[i][j] = matrix1[i][j] + matrix2[i][j];
        }
    }
}

void display(int r, int c, int matrix[r][c]) {
    printf("Output Matrix:\n");
    for (int i = 0; i < r; i++) {
        for (int j = 0; j < c; j++) {
            printf("%d\t", matrix[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int r, c;
    printf("Enter the number of rows and columns of the matrices:\n");
    scanf("%d %d", &r, &c);
    if (r > 0 && c > 0) {

        int matrix1[r][c], matrix2[r][c], result[r][c];
        input(r, c, matrix1, "1st");
        input(r, c, matrix2, "2nd");

        calculate(r, c, matrix1, matrix2, result);
        display(r, c, result);
    } else {
        printf("Error: Rows and columns should be greater than 0\n");
    }
    return 0;
}
